t/t0410/read-object

```diff
@@ -102,7 +102,7 @@ while (1) {
 	my ($command) = packet_txt_read() =~ /^command=([^=]+)$/;
 
 	if ( $command eq "get" ) {
-		my ($sha1) = packet_txt_read() =~ /^sha1=([0-9a-f]{40})$/;
+		my ($sha1) = packet_txt_read() =~ /^sha1=([0-9a-f]{40,64})$/;
```

Comment:
It's hard to tell what's going on in this snippet, so I'll ask a maybe dumb question.

The right hand side has `/^sha1=` in the pattern.  Does that pattern need to change too?
That is, would whatever is generating this token stream send a line with `sha1` and a 64 byte hash
or would it be either /sha1=..40 || /sha256=..64 or something like that ??

Again, not sure the context here, so just asking.

Reply:
Good point. This is actually a hard-coded part of the `read-object` "protocol": https://github.com/microsoft/git/blob/22fe9884d70b62e7197a915597afe11cbe9ad329/sha1-file.c#L1030

The question now is whether we really want to be precise and change that part of the protocol. That would mean that the VFSforGit code would have to change accordingly... I'm not sure that I want to open that can of worms.

Yes, sure, technically it would be incorrect to ask for a SHA-256 object via `sha1=<object-id>`. But does it hurt? And it sure would make maintenance less burdensome to leave it as-is.

Reply:
agreed.  i don't think we should change it just to be changing it.  i'm more worried about someone seeing that line in sha1-file.c and deciding that it needs to be "fixed" somehow -- or if in fact someone already had fixed it and the test suite had insufficient coverage.

Perhaps, we could add a comment in t0410 summarizing this.

Changing the test to be something like `=~ /^sha[0-9]+=...` might help future-proof us, but then again, we stick the result in a variable called `sha1`, so we're not really making anything clearer.  A comment should be sufficient for now.

Reply:
I guess the best idea would be to _change_ the `sha1` (or more likely, add an alias) to `hash` or even `oid`.

Reply:
yeah, renaming the variable name to something like `oid` would be fine.   Whatever is easiest -- I'm really not trying to cause extra work here.   :-)

Reply:
It's not really a variable. It's a hard-coded part of the `read-object` protocol, i.e. the text that is sent to the hook to request certain objects.